components/wpt-results.html

```diff
@@ -248,64 +218,70 @@
       }
 
       _queryChanged () {
-        this._refreshDisplayedSpecDirs()
+        this._refreshDisplayedNodes()
       }
 
-      _refreshDisplayedSpecDirs () {
+      _refreshDisplayedNodes () {
         /* Recomputes the list of displayed spec directories and their child test results. */
-        const matchesQuery = (tf) => tf.testFile.toLowerCase().includes(this.query.toLowerCase())
-        const displayedSpecDirs = []
-
-        for (var key in this.specDirs) {
-          let specDir = this.specDirs[key]
-          let displaySpecDir = {}
-
-          displaySpecDir.specName = specDir.specName
-          displaySpecDir.results = this.testRuns.map(testRun => specDir.results[testRun.results_url])
+        const displayedNodeDict = {}
```

Comment:
Great suggestion, I simplified it to `displayedNodes` and changed it to a `Map`.